gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/integration/use_case/IngestIntegrationApisUseCase.java

```diff
@@ -90,50 +96,116 @@ public Completable execute(Input input) {
-                    }
-                };
-            })
-            .forEach(page -> createApiDocumentationDomainService.createPage(page, auditInfo));
+    private void createApi(Api federatedApi, IntegrationApi integrationApi, PrimaryOwnerEntity primaryOwner, AuditInfo auditInfo) {
+        try {
+            createApiDomainService.create(federatedApi, primaryOwner, auditInfo, validateFederatedApi::validateAndSanitizeForCreation);
+
+            Optional
+                .ofNullable(integrationApi.plans())
+                .ifPresent(plans ->
+                    plans
+                        .stream()
+                        .map(p -> PlanModelFactory.fromIntegration(p, federatedApi))
+                        .forEach(p -> createPlanDomainService.create(p, List.of(), federatedApi, auditInfo))
+                );
+
+            Optional
+                .ofNullable(integrationApi.pages())
+                .ifPresent(pages ->
+                    pages
+                        .stream()
+                        .flatMap(page -> buildPage(page, integrationApi, federatedApi.getId()))
+                        .forEach(page -> createApiDocumentationDomainService.createPage(page, auditInfo))
+                );
+        } catch (Exception e) {
+            log.warn("An error occurred while importing api {}", federatedApi, e);
+        }
+    }
+
+    private void updateApi(Api federatedApi, IntegrationApi integrationApi, AuditInfo auditInfo, PrimaryOwnerEntity primaryOwner) {
+        log.debug("API already ingested [id={}] [name={}], performing update", federatedApi.getId(), federatedApi.getName());
+        try {
+            /*
+             * Because we use page names as identifiers we need to look for the page by existing API name.
+             * Otherwise, in case of an API name update, we would create a new page instead of updating the existing one.
+             * If we move on to some predictable unique ID the existingApiName will become unnecessary.
+             */
+            var existingApiName = apiCrudService.get(federatedApi.getId()).getName();
+
+            updateFederatedApiDomainService.update(federatedApi, auditInfo, primaryOwner);
+
+            Optional
+                .ofNullable(integrationApi.plans())
+                .ifPresent(plans ->
+                    plans
+                        .stream()
+                        .map(p -> PlanModelFactory.fromIntegration(p, federatedApi))
+                        .forEach(p ->
+                            planCrudService
+                                .findById(p.getId())
+                                .ifPresentOrElse(
+                                    existingPlan -> updatePlanDomainService.update(p, List.of(), null, auditInfo),
+                                    () -> createPlanDomainService.create(p, List.of(), federatedApi, auditInfo)
+                                )
+                        )
+                );
```

Comment:
```suggestion
            Optional
                .ofNullable(integrationApi.plans())
                .stream()
                .flatMap(plans -> plans.stream())
                .map(p -> PlanModelFactory.fromIntegration(p, federatedApi))
                .forEach(p ->
                    planCrudService
                        .findById(p.getId())
                        .ifPresentOrElse(
                            existingPlan -> updatePlanDomainService.update(p, List.of(), null, auditInfo),
                            () -> createPlanDomainService.create(p, List.of(), federatedApi, auditInfo)
                        )
                );
```

I think that you can reduce indentation with something like this.

Reply:
Sure, good idea. I improved other streams as well :)